experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -107,6 +107,12 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			logger.Error(err, "Unable to create Plan Object")
 			return ctrl.Result{}, err
 		}
+
+		// Read after create to get the UID
+		if err := r.Client.Get(ctx, planNN, &plancr); err != nil {
```

Comment:
Do you know for a fact this isn't set?  I ask because I thought client.Client actually populated back some values, like resourceVersion in particular.

Reply:
I dont know for a fact. I assumed it did not. Will check and remove this if that is indeed the case.